src/main/java/org/cactoos/map/MapEnvelope.java

```diff
@@ -170,7 +172,12 @@ public final int hashCode() {
                     ).value();
                     return new SumOfInt(
                         () -> 37 * keys,
-                        () -> entry.getValue().hashCode()
+                        () -> new Ternary<>(
+                            entry.getValue(),
+                            o -> o != null,
+                            o -> o.hashCode(),
+                            o -> 0
+                        ).value()
```

Comment:
@Umbrah here you should be able to simplify the code in the same way as in ùy other comment on `contentsEqual`.